src/routes/news/containers/WithCustomNewsProvider.js

```diff
@@ -261,46 +236,35 @@ const withCustomNewsProvider = <
         }
       }
 
-      renderHeader = (cityModel: CityModel) => {
+      renderHeader = (cityModel?: CityModel) => {
         const { selectedNewsType } = this.state
         return (
           <HeaderContainer>
-            {newsTabs.map(tab =>
-              cityModel[tab.toggleAttr] ? (
-                <TranslatedNewsTypeItem
-                  key={tab.type}
-                  tab={tab}
-                  selectedNewsType={selectedNewsType}
-                  onItemPress={this.selectNewsItemAndScrollToTop}
-                />
-              ) : null
-            )}
-          </HeaderContainer>
-        )
+            {cityModel && cityModel.pushNotificationsEnabled &&
+              <TranslatedNewsTypeItem key='pushNotificationsEnabled' tab={newsTabs[0]} selectedNewsType={selectedNewsType}
+                                  onItemPress={this.selectNewsItemAndScrollToTop} />}
+            {cityModel && cityModel.tunewsEnabled &&
+              <TranslatedNewsTypeItem key='tunewsEnabled' tab={newsTabs[1]} selectedNewsType={selectedNewsType}
+                                  onItemPress={this.selectNewsItemAndScrollToTop} />}
+          </HeaderContainer>)
```

Comment:
Not sure if I'd expect a provider to already render stuff. Maybe you could simplify/extract logic from this component since it's also quite big.